app/common/data/interfaces/collections.py

```diff
@@ -86,6 +86,31 @@ def update_submission_data(submission: Submission, question: Question, data: All
     return submission
 
 
+def get_all_submissions_with_mode_for_collection_with_full_schema(
+    collection_id: UUID, submission_mode: SubmissionModeEnum
+) -> ScalarResult[Submission]:
+    """
+    Use this function to get all submission data for a collection - it
+    loads all the question/expression/user data at once to optimise
+    performance and reduce the number of queries compared to looping
+    through them all individually.
+    """
+    return db.session.scalars(
+        select(Submission)
+        .where(Submission.collection_id == collection_id)
+        .where(Submission.mode == submission_mode)
+        .options(
+            joinedload(Submission.collection)
+            .selectinload(Collection.sections)
+            .selectinload(Section.forms)
+            .selectinload(Form.questions)
+            .selectinload(Question.expressions),
+            joinedload(Submission.events),
```

Comment:
Ah I missed the commas earlier. We can switch this to a selectinload as long as we remove the comma above. The fact these are chaining together is not very obvious with those sneaky commas sporadically dotted in.

Reply:
Nice! Fixed